editor/src/components/canvas/canvas-strategies/shared-keyboard-strategy-helpers.ts

```diff
@@ -93,54 +93,58 @@ export function getKeyboardStrategyGuidelines(
   canvasState: InteractionCanvasState,
   interactionState: InteractionSession,
   draggedFrame: CanvasRectangle,
-): Array<GuidelineWithSnappingVector> {
+): Array<GuidelineWithSnappingVectorAndPointsOfRelevance> {
   const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
   const moveGuidelines = collectParentAndSiblingGuidelines(
     interactionState.metadata,
     selectedElements,
-  )
+  ).map((g) => g.guideline)
+
   const { horizontalPoints, verticalPoints } = Utils.getRectPointsAlongAxes(draggedFrame)
-  const closestGuideLines: Array<GuidelineWithSnappingVector> = mapDropNulls((guideline) => {
-    switch (guideline.type) {
-      case 'XAxisGuideline': {
-        const snappingVector = Guidelines.getOffsetToSnapToXGuideline(
-          horizontalPoints,
-          guideline,
-          null,
-        )
-        if (Utils.magnitude(snappingVector) === 0) {
-          return {
-            guideline: guideline,
-            snappingVector: snappingVector,
-            activateSnap: true,
+  const closestGuideLines: Array<GuidelineWithSnappingVectorAndPointsOfRelevance> = mapDropNulls(
+    (guideline) => {
+      switch (guideline.type) {
+        case 'XAxisGuideline': {
+          const snappingVector = Guidelines.getOffsetToSnapToXGuideline(
+            horizontalPoints,
+            guideline,
+            null,
+          )
+          if (Utils.magnitude(snappingVector) === 0) {
+            return {
+              guideline: guideline,
+              snappingVector: snappingVector,
+              pointsOfRelevance: [],
+            } as GuidelineWithSnappingVectorAndPointsOfRelevance
```

Comment:
Please can you use a factory function or some other helper here rather than casting. We've found that it makes refactoring much easier.

Reply:
done!